pgctl/cli.py

```diff
@@ -228,10 +228,21 @@ def __change_state(self, state):
         # If we're starting a service, run the playground-wide "pre-start" hook (if it exists).
         # This is intentionally done without holding a lock, since this might be very slow.
         if state is Start:
-            self.run_pre_start_hook()
+            self._run_playground_wide_hook('pre-start')
 
         with self.playground_locked():
-            return self.__locked_change_state(state)
+            failures = self.__locked_change_state(state)
+
+        # If the playground is in a fully stopped state, run the playground wide
+        # post-stop hook. As with pre-start, this is done without holdinga lock.
+        if state is Stop:
+            for service in self.all_services:
+                if self._get_service_state(service)['state'] != 'down':
+                    break
```

Comment:
it might be better (less race-y) to get the service states while still holding the lock?